### PR TITLE
Wandb

### LINKED ISSUE #5: Training logs (Tensorboard or WandB)
Would be great to have a way to monitor learning progress.

### PR DESCRIPTION
**Pull request by Olivier-tl**
- Added Logging to WANDB 📈📊
- You can see an example of a run here : https://wandb.ai/game-theory/MO-DAGAN
- @Akssi & @wwwnev you have been invited by email to the game-theory team on WandB

Closes #5

### CODE REVIEW

File: trainers/gan_trainer.py [hunk 1]
```diff
@@ -96,8 +96,8 @@ def train(self):
             g_cost = -g_loss
             self.g_optimizer.step()
 
-            # TODO: Log to WandB
-            # logger.info(f'Generator iteration: {g_iter}/{self.model.generator_iters}, g_loss: {g_loss}')
+            # Log to WandB
+            wandb.log({'g_loss': g_loss})
```
Comment (Akssi) [inline trainers/gan_trainer.py:100]: I think we have to either log all at once or use the step argument so the x axis is the epochs/batch
see https://docs.wandb.ai/library/log#incremental-logging

Comment (Olivier-tl) [reply trainers/gan_trainer.py:100]: I tried to use the step argument actually! If you look at commit my commit history, the last commit removes the step argument because the logging was ignored by WandB since the steps were not strictly going up. (for example the steps of the discriminator loops.) Not sure what do to here.

**Comment by Olivier-tl** [comment]
You can see an example of logging for the GAN here along with a better sample (4500 iters) here : https://wandb.ai/game-theory/MO-DAGAN/runs/1bb2pb26

File: trainers/gan_trainer.py (hunk 1 above)
Comment (Olivier-tl) [reply trainers/gan_trainer.py]: Ok now i'm directly logging the generator and discriminator iterations! I think it's the best way to go.

Comment (Olivier-tl) [reply trainers/gan_trainer.py]: Result with that here : https://wandb.ai/game-theory/MO-DAGAN/runs/u3cyc391

**Review by wwwnev — APPROVED** [review]
